sdk/Sdk.Generators/ExtensionStartupRunnerGenerator.cs

```diff
@@ -0,0 +1,168 @@
+                                                        .FirstOrDefault(a => a.AttributeClass?.Name == attributeTypeName &&
+                                                                        //Call GetFullName only if class name matches.
+                                                                        a.AttributeClass.GetFullName() == attributeTypeFullName);
+                if (extensionStartupAttribute != null)
+                {
+                    // WorkerExtensionStartup has a constructor with one parameter, the type of startup implementation class.
+                    TypedConstant firstConstructorParam = extensionStartupAttribute.ConstructorArguments[0];
+                    if (firstConstructorParam.Value is ITypeSymbol typeSymbol)
+                    {
+                        var fullTypeName = typeSymbol.ToDisplayString();
+                        typeNames.Add(fullTypeName);
+                    }
+                }
+            }
+
+            return typeNames;
+        }
+
+        /// <summary>
+        /// Writes an assembly attribute with type information about our auto generated WorkerExtensionStartupCodeExecutor class.
+        /// </summary>
+        private static void WriteAssemblyAttribute(IndentedTextWriter textWriter)
+        {
+            textWriter.WriteLine(
+                "[assembly: WorkerExtensionStartupCodeExecutorInfo(typeof(Microsoft.Azure.Functions.Worker.WorkerExtensionStartupCodeExecutor))]");
+        }
+
+        /// <summary>
+        /// Writes a class with code which calls the Configure method on each implementation of participating extensions.
+        /// We also have it implement the same "IWorkerExtensionStartup" interface which extension authors implement.
+        /// </summary>
+        private static void WriteStartupCodeExecutorClass(IndentedTextWriter textWriter, IEnumerable<string> typeNames)
+        {
+            textWriter.WriteLine("internal class WorkerExtensionStartupCodeExecutor : IWorkerExtensionStartup");
+            textWriter.WriteLine("{");
+            textWriter.Indent++;
+            textWriter.WriteLine("public void Configure(IFunctionsWorkerApplicationBuilder applicationBuilder)");
+            textWriter.WriteLine("{");
+            textWriter.Indent++;
+
+            foreach (var fullyQualifiedTpeName in typeNames)
+            {
+                textWriter.WriteLine("try");
+                textWriter.WriteLine("{");
+                textWriter.Indent++;
+                textWriter.WriteLine($"new {fullyQualifiedTpeName}().Configure(applicationBuilder);");
+                textWriter.Indent--;
+                textWriter.WriteLine("}");
+                textWriter.WriteLine("catch (Exception ex)");
+                textWriter.WriteLine("{");
+                textWriter.Indent++;
+                textWriter.WriteLine($"Console.WriteLine(\"Error calling Configure on {fullyQualifiedTpeName} instance.\"+ex.ToString());");
```

Comment:
I created #850 to implement the proper startup logger.

Reply:
My thought process for wrapping each call in its own `try-catch` was to prevent one bad extension update/upgrade crashing the entire app. This current approach is more of a degraded experience (the bad extensions startup code won't be executed, but rest of the extension's startup will still work) of the app instead of a crashed app. 

Will discuss this item in the next worker triage/design sync.
Let's